### PR TITLE
Changed maven deploy configuration

### LINKED ISSUE #230: During release fat jar is uploaded instead of diktat-rules
`https://central.artipie.com/akuleshov7/diktat/org/cqfn/diktat/diktat-rules/1.0.2/diktat-rules-1.0.2.jar` is a fat jar (size ~60MB), and should be just diktat-rules instead.

Probably, something wrong with `maven-assembly-plugin` config.

### PR DESCRIPTION
**Pull request by petertrr**
This closes #230 

Fat jar has been installed/deployed twice under different names. This is because artifacts under same artifactId can differ only in classifiers. If we want separate `diktat-$version.jar`, we should create separate pom for it.

Overall the changes are as follows:
* New maven module `diktat-ruleset` is introduced
* It produces a single artifact, jar with dependencies (excluding all kotlin-related dependencies, which are provided by `ktlint` executable)
* The new artifact is called `diktat-ruleset-$version.jar`, it is uploaded to github releases.

### CODE REVIEW

**Comment by petertrr** [comment]
@akuleshov7 do we still need separate `diktat.jar`, or it is fine to upload `diktat-rules-jar-with-dependencies`?

File: diktat-rules/pom.xml [hunk 1]
```diff
@@ -168,7 +168,7 @@
                         <configuration>
                             <artifacts>
                                 <artifact>
-                                    <file>${build.directory}/diktat.jar</file>
+                                    <file>${build.directory}/diktat-${project.version}-jar-with-dependencies.jar</file>
```
Comment (akuleshov7) [inline diktat-rules/pom.xml:171]: we really need such long name?

Comment (petertrr) [reply diktat-rules/pom.xml:171]: I believe it's standard classifier for fat jars. If we want to deploy fat jat simply as `diktat-${version}.jar`, I can add this configuration as a separate maven module.

Comment (akuleshov7) [reply diktat-rules/pom.xml:171]: standard pattern is fat-jar-with-dependencies, but I guess `diktat-$version-fat.jar` is absolutely fine and more usable

**Review by akuleshov7 — APPROVED** [review]
Lgtm, let’s try

**Pull request merged**